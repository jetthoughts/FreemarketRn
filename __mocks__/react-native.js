const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
