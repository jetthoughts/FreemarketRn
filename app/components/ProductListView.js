import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ListView, Text } from 'react-native';

import ProductListRow from '../components/ProductListRow';

export default class ProductListView extends Component {
  constructor(props) {
    super(props);

    const ds = new ListView.DataSource({
      rowHasChanged: (r1, r2) => r1.id !== r2.id,
    });
    this.dataSource = ds.cloneWithRows(props.products);
  }

  renderRow(row) {
    const { name, image, price } = row;
    return <ProductListRow {...{ name, image, price }} />;
  }

  render() {
    return (
      <ListView dataSource={this.dataSource} renderRow={(row) => this.renderRow(row)} />
    );
  }
}

ProductListView.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
  })).isRequired,
};

// export default class ProductListView extends Component {
//   constructor(props) {
//     super(props);
//
//     const ds = new ListView.DataSource({
//       rowHasChanged: (r1, r2) => r1.id !== r2.id,
//     });
//     this.dataSource = ds.cloneWithRows([props.products]);
//   }
//
//   renderRow(row) {
//     const { name, image, price } = row;
//     return <ProductListRow {...{ name, image, price }} />;
//   }
//
//   redner() {
//     return (
//       <Text>wtf2</Text>
//     );
//   }
// }
//
// ProductListView.propTypes = {
//   products: PropTypes.arrayOf(PropTypes.shape({
//     id: PropTypes.number.isRequired,
//     name: PropTypes.string.isRequired,
//     image: PropTypes.string.isRequired,
//     price: PropTypes.number.isRequired,
//   })).isRequired,
// };
