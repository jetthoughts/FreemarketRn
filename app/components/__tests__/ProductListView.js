import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProductListView from '../ProductListView';

it('renders correctly', () => {
  const category = { name: 'Home Decor' };
  const tree = renderer.create(<ProductListView category={category} />).toJSON();

  expect(tree).toMatchSnapshot();
});
