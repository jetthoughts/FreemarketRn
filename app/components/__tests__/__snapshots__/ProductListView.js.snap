// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 0,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={
    Object {
      "backgroundColor": "dimgray",
    }
  }
>
  <View />
</RCTScrollView>
`;
